instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -50,7 +50,13 @@ def query(sql)
               ),
               kind: :client
             ) do |_span, context|
-              propagator.inject(sql, context: context)
+              if propagator
+                dup = sql.frozen?
+                sql = sql.dup if dup
+                propagator.inject(sql, context: context)
+                sql.freeze if dup
+              end
```

Comment:
This is kinda messy and hard to follow. It'd be clearer to:
```suggestion
              if propagator
                injectable_sql = +sql
                propagator.inject(injectable_sql, context: context)
                sql = -injectable_sql if sql.frozen?
              end
```
or
```suggestion
              if propagator && sql.frozen?
                sql = +sql
                propagator.inject(sql, context: context)
                sql.freeze
              elsif propagator
                propagator.inject(sql, context: context)
              end
```